sentinel-core/src/main/java/com/alibaba/csp/sentinel/util/ConfigUtil.java

```diff
@@ -108,18 +113,11 @@ private static Properties loadPropertiesFromClasspathFile(String fileName) {
         Properties properties = new Properties();
         for (URL url : list) {
             try {
-                Properties p = new Properties();
-                InputStream input = url.openStream();
-                if (input != null) {
-                    try {
-                        p.load(input);
-                        properties.putAll(p);
-                    } finally {
-                        try {
-                            input.close();
-                        } catch (Throwable t) {
-                        }
-                    }
+                try (BufferedReader bufferedReader =
+                             Files.newBufferedReader(Paths.get(url.toURI()), Charset.forName(SentinelConfig.charset()))) {
```

Comment:
It seems that `Files.newBufferedReader(xxx)` only support `file` URI type. When the properties file is located in the jar (i.e. `jar:file://xxx`), exceptions will be thrown:

```
java.nio.file.FileSystemNotFoundException
        at com.sun.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:171)
        at com.sun.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:157)
        at java.nio.file.Paths.get(Paths.java:143)
```

Maybe we could just wrap the `url.openStream()` with `InputStreamReader` like this:

```java
for (URL url : list) {
    Properties p = new Properties();
    try (InputStreamReader reader = new InputStreamReader(url.openStream(),
        Charset.forName(SentinelConfig.charset()))) {
        p.load(reader);
        properties.putAll(p);
    } catch (Throwable e) {
        e.printStackTrace();
    }
}

```

Reply:
Thanks for your review, I made a stupid mistake and now I am correcting it.

Reply:
👍